src/colvarproxy_io.h

```diff
@@ -0,0 +1,157 @@
+  {
+    return input_prefix_str;
+  }
+
+  /// Default prefix to be used for all output files (final configuration)
+  inline std::string & output_prefix()
+  {
+    return output_prefix_str;
+  }
+
+  /// Prefix of the restart (checkpoint) file to be written next
+  inline std::string & restart_output_prefix()
+  {
+    return restart_output_prefix_str;
+  }
+
+  /// Default restart frequency (as set by the simulation engine)
+  inline int default_restart_frequency() const
+  {
+    return restart_frequency_engine;
+  }
+
+  /// Buffer from which the input state information may be read
+  inline char const * & input_buffer()
+  {
+    return input_buffer_;
+  }
+
+  /// Returns a reference to given input stream, creating it if needed
+  /// \param input_name File name (later only a handle)
+  /// \param description Purpose of the file
+  /// \param error_on_fail Raise error when failing to open (allow testing)
+  virtual std::istream *input_stream(std::string const &input_name,
+                                     std::string const description = "file/channel",
+                                     bool error_on_fail = true);
+
+  /// Returns a reference to given input stream, null otherwise
+  virtual std::istream *get_input_stream(std::string const &input_name);
+
+  /// Closes the given input stream
+  virtual int close_input_stream(std::string const &input_name);
+
+  /// \brief Returns a reference to the given output channel;
+  /// if this is not open already, then open it
+  virtual std::ostream *output_stream(std::string const &output_name,
+                                      std::ios_base::openmode mode =
+                                      std::ios_base::out);
+
+  /// Returns a reference to output_name if it exists, NULL otherwise
+  virtual std::ostream *get_output_stream(std::string const &output_name);
+
+  /// \brief Flushes the given output channel
+  virtual int flush_output_stream(std::ostream *os);
+
+  /// \brief Flushes all output channels
+  virtual int flush_output_streams();
+
+  /// \brief Closes the given output channel
+  virtual int close_output_stream(std::string const &output_name);
+
+protected:
+
+  /// Prefix of the input state file to be read next
+  std::string input_prefix_str;
+
+  /// Default prefix to be used for all output files (final configuration)
+  std::string output_prefix_str;
+
+  /// Prefix of the restart (checkpoint) file to be written next
+  std::string restart_output_prefix_str;
+
+  /// How often the simulation engine will write its own restart
+  int restart_frequency_engine;
+
+  /// \brief Currently opened output files: by default, these are ofstream objects.
+  /// Allows redefinition to implement different output mechanisms
+  std::list<std::ostream *> output_files;
+  /// \brief Identifiers for output_stream objects: by default, these are the names of the files
+  std::list<std::string>    output_stream_names;
+
+  std::map<std::string, std::istream *> input_streams_;
```

Comment:
Why are only input streams stored in `std::map`, while output streams are stored separately in two `std::list`s?

Reply:
The latter is much older, when we couldn't be sure that `std::map` would be available on so many exotic platforms. I plan on converting the latter to `std::map` as well.

Reply:
I am OK if you do not have time to unify the handling of `std::istream*` and `std::ostream*` at this time.